docs/design/mint-lock.md

```diff
@@ -0,0 +1,101 @@
+
+This endpoint is not permissioned. It is possible for anyone to pay for the creation of a new ERC20 representing a Cosmos asset, but it is up to the validators and the Gravity Cosmos module to declare any given ERC20 as the representation of a given asset.
+
+When a user on Ethereum calls `deployERC20` they pass arguments describing the desired asset. [Gravity.sol](/solidity/contracts/Gravity.sol) uses an ERC20 factory to deploy the actual ERC20 contract using a known good code and assigns ownership of the entire balance of the new token to itself before firing a `ERC20DeployedEvent`
+
+The validators oracle processes observe this event and decide if a Cosmos asset has been accurately represented (correct decimals, correct name, no existing representation). If this is the case the ERC20 contract address is adopted and stored as the definitive representation of that Cosmos asset on Ethereum.
+
+For further details on oracle operation see [oracle](/docs/design/oracle.md)
+
+From this point on the process is nearly identical to the Cosmos -> Ethereum process for Ethereum originated assets.
+
+The user sends a `MsgSendToEth` that specifies a fee denominated in the token they are sending across the bridge. (for the
+reasoning view the [transaction batch rewards](/docs/design/mint-and-lock.md/###transaction-batch-rewards))
+
+At some point in the future a relayer will determine that the pool contains enough transactions to bother executing a batch of that type and relay it to Ethereum. For a discussion of edge cases here see the [batch creation spec](/spec/batch-creation-spec.md)
+
+When the batch transaction executes on the Ethereum chain the ERC20 token will be sent out the bridge to the Ethereum destination specified by the user in `MsgSendToEth`. An Ethereum event `TransactionBatchExecutedEvent` will be fired and picked up by the validators oracle processes. Once the resulting `MsgWithdrawClaim` has passed the Cosmos asset will be locked in the Gravity module to later be unlocked once it is returned to Cosmos.
+
+The tl;dr of this process is that the 'minting' of Cosmos assets on Ethereum is done all at once and the total balance simply exists in the [Gravity.sol](/solidity/contracts/Gravity.sol) contract at all times, which 'mints' them by sending them out and 'burns' them by locking them up again.
+
+As a related consequence this means that the supply of a given Ethereum asset on Cosmos is correct, but the supply of a given Cosmos asset on Ethereum is the total number of tokens minus the number in the Gravity bridge address.
+
+### Ethereum -> Cosmos
+
+This is identical to Ethereum -> Cosmos for Ethereum based assets. Please see that section and refer to the above section for details on how the overall process differs for Cosmos assets.
+
+## Relaying rewards
```

Comment:
should this be added into the rewards section?

Reply:
it's intended to be the EThereum -> Cosmos flow for Cosmos based assets but I wasn't sure how much duplication was really needed.